src/playback-watcher.js

```diff
@@ -211,6 +211,24 @@ export default class PlaybackWatcher {
       this.timer_ = setTimeout(this.skipTheGap_.bind(this),
                                difference * 1000,
                                currentTime);
+      return;
+    }
+
+    let currentRange = Ranges.findRange(buffered, currentTime);
+
+    // Sometimes the player can stall for unknown reasons within a contiguous buffered
+    // region with no indication that anything is amiss (seen in Firefox). Seeking to
+    // currentTime is usually enough to kickstart the player.
+    if (currentRange.length) {
```

Comment:
Do we want to guard against cases where we're at the end of the buffered region and just waiting for data?